### ISSUE TITLE
gexport not exporting Periodically, Every Polling Cycle

### ISSUE THREAD

**Issue by TheMetMan** [Cacti/plugin_gexport#17]
I have a fresh install of Cacti 1.1.16 on CentOS 7. (The upgrade using Yum from 0.8x to 1.1.12 failed).
I can export the graphs manually using the 'Export Now' option, that works fine, but the 'Next start' is in the past, and does not update, even if I change the Polling Method to Hourly, then back again.
Set to 'Hourly' or 'Periodic' it went past the time it said as the 'Next Start' without updating.
So basically it can do what it is meant to do, but does not.
Can you check this please?
Thanks
PS I have raised this on the Cacti Forum, but nothing back yet.

**Comment by cigamit** [comment]
What is your poller interval/frequency?

**Comment by cigamit** [comment]
No feedback.

**Comment by TheMetMan** [comment]
My polling interval is every 5 mins, and I have 9 items

**Comment by cigamit** [comment]
Welcome back from vacation then.  Can you attach a screen shot of the gexport table view for the Cacti UI?

**Comment by TheMetMan** [comment]
Thanks for responding.
I have Enable Automatic Graph Creation checked, the graphs are being created, and the export works manually.
Attached is the screenshot.
Regards
Francis

----- Original Message -----

**Comment by TheMetMan** [comment]
Dear Jimmy, sorry if you get this twice, I sent it last night from the wrong Email address.
Thanks for responding.
I have Enable Automatic Graph Creation checked, the graphs are being created, and the export works manually.
Attached is the screenshot.
Regards
Francis



----- Original Message -----

**Comment by cigamit** [comment]
Sorry, you did not actually attach a screen shot.  Try again.

**Comment by TheMetMan** [comment]
OK another try, but it will not let me paste the file here. I will have to upload it somewhere for you to download, or Email directly to you in some way.

**Comment by cigamit** [comment]
The way to save files to the issue log is to save them to a physical file first, and then drag and drop the files to the github issue text area.  Works in all Windows browsers that way anyway.

**Comment by cigamit** [comment]
Also, table view and not edit view.

**Comment by TheMetMan** [comment]
OK, here it is.
![gexporttableview](https://user-images.githubusercontent.com/6077253/29762504-affeab06-8bc6-11e7-8796-36051b8fb7e2.png)
Drag & Drop in Firefox on Linux

**Comment by cigamit** [comment]
Okay, seeing is believing.  I was suspending disbelief until I saw the image.  The 21st is a long while ago ;)

**Comment by cigamit** [comment]
Can you please paste the results of the following:

`SELECT * FROM graph_exports\G`

The code is pretty simple, if next start is less than the current timestamp, then start.

**Comment by TheMetMan** [comment]
OK, here is the output
*************************** 1. row ***************************
                     id: 1
                   name: Export All Graphs
            export_type: local
                enabled: on
    export_presentation: preview
  export_effective_user: 1
    export_expand_hosts: off
           export_theme: modern
             graph_tree: 1
             graph_site: 1
           graph_height: 100
            graph_width: 300
       graph_thumbnails: 
          graph_columns: 2
          graph_perpage: 20
              graph_max: 2000
       export_directory: /var/www/html/cacti_graphs
  export_temp_directory: /tmp/
          export_timing: periodic
            export_skip: 1
          export_hourly: 00
           export_daily: 00:00
 export_sanitize_remote: 
            export_host: 
            export_port: 
         export_passive: 
            export_user: 
        export_password: 
export_private_key_path: 
                 status: 0
             export_pid: 0
             next_start: 2017-08-21 19:07:00
           last_checked: 2017-08-22 07:30:09
           last_started: 2017-08-22 07:30:09
             last_ended: 2017-08-22 07:30:11
           last_errored: 0000-00-00 00:00:00
           last_runtime: 1.8305959701538
             last_error: NULL
           total_graphs: 8
1 row in set (0.00 sec)

**Comment by cigamit** [comment]
Okay, this may sound silly, but can you goto Console > Configuration > Plugin Manager and see if gexport is enabled?

**Comment by TheMetMan** [comment]
It has a little green icon, not a red one
On 29 Aug 2017 12:34, Jimmy Conner <notifications@github.com> wrote:Okay, this may sound silly, but can you goto Console > Configuration > Plugin Manager and see if gexport is enabled?

**Comment by TheMetMan** [comment]
Dear Jimmy, I think I have to eat some Humble Pie, and thank you for your patience with me.

I have a green icon with a tick in the Plugin Management. If I hover over this It says 'Enable Plugin', so I click it, it turns red and hovering says 'Disable Plugin'. I thought that if it was green then it was enabled, if red, disabled, but I find it is the opposite. Since doing this it is working, so all along it is my error in misunderstanding the icons.

Now going to the Graph Exports under Utilities, I can select an action 'Enable', having checked the right hand box, and then click 'Go' I get a message 'Click 'Continue' to disable the following Graph Export Definition.' , this is strange! I would have thought if I selected 'Enable' it would enable the export. Similarly if I select 'Disable' I get 'Click 'Continue' to enable the following Graph Export Definition.' which is not what I would expect.

I think this is part of my confusion.

Anyway it does seem to be working now. I am REALLY grateful for your patience and persistence in helping sort this out.
Regards
Francis



----- Original Message -----

**Comment by cigamit** [comment]
No problem Francis.  Glad we figured it out.
